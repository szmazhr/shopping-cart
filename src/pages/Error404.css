.error-404{
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  height: 80vh;
}
.error-404 h1{
  font-size: 2.5rem;
  font-weight: 700;
  color: rgb(26, 26, 26);
  margin-bottom: 0;
}
.error-404 .btn{
  background-color: rgb(139, 218, 191);
  display: inline-block;
  font-size: 1.2rem;
  padding: 0.8rem 1.5rem;
  color: rgb(26, 26, 26);
}