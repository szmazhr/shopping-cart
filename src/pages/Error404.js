import { Link } from 'react-router-dom';
import './Error404.css';
function Error404() {
  return (
    
    <div className="error-404">
      <h1>404</h1>
      <p>Page not found</p>
      <Link className='btn' to='/'>Home</Link>
    </div>
  )
}

export default Error404