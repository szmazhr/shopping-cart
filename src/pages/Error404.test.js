import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Error404 from './Error404';

describe('Error404', () => {
  it('renders correctly', () => {
    const {asFragment} = render(<Error404/>, { wrapper: MemoryRouter });
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByRole('heading')).toHaveTextContent('404');
    expect(screen.getByRole('link')).toHaveAttribute('href', '/');
  });
});