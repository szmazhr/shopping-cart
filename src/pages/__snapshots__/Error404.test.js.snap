// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error404 renders correctly 1`] = `
<DocumentFragment>
  <div
    class="error-404"
  >
    <h1>
      404
    </h1>
    <p>
      Page not found
    </p>
    <a
      class="btn"
      href="/"
    >
      Home
    </a>
  </div>
</DocumentFragment>
`;
